apps/examples/webclient/webclient_main.c

```diff
@@ -245,21 +304,25 @@ int webclient_init_request(void *arg, struct http_client_request_t *request)
 			if (t > 0 && t <= WEBCLIENT_CONF_MAX_ENTITY_SIZE) {
 				request->entity = (char *)malloc(t);
 				if (request->entity == NULL) {
-					return -1;
+					goto exit;
 				}
 				g_testentity = 1;
 				memset(request->entity, '1', t);
 			} else {
 				printf("entity is too big\n");
-				return -1;
+				goto exit;
 			}
 		} else {
-			return -1;
+			goto exit;
 		}
 	}
 
 	request->buflen = WEBCLIENT_BUF_SIZE;
-	return 0;
+	ret = 0;
+exit:
+	WEBCLIENT_FREE_INPUT(input, input->argc);
```

Comment:
How about free this at main function?
There are two reasons.
1. allocation and free locations are at same thread.
2. Due to pthread_join, main wll wait until finishing callback function so that we can add it at after pthread_join.

Reply:
it is ok to release simple memory at other thread. And we should consider how long does it hold input memory not used anymore